cert/stdlib/cert_auth_stdlib.go

```diff
@@ -0,0 +1,157 @@
+}
+
+// NewCA initializes a Certificate Authority.
+func NewCA() *CA {
+	return &CA{}
+}
+
+// SetCACert sets the CA certificate with the provided certificate and key.
+func (ca *CA) SetCACert(cert *cert.Certificate) error {
+	var err error
+
+	// PEM to DER
+	pbCert, _ := pem.Decode(cert.Cert)
+
+	// parse the Certificate
+	ca.cert, err = x509.ParseCertificate(pbCert.Bytes)
+	if err != nil {
+		return err
+	}
+
+	// Parse the PrivateKey
+	ca.key, err = ssh.ParseRawPrivateKey(cert.Key)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GenerateCACert generates a CA certificate, key and CSR based on the provided input.
+func (ca *CA) GenerateCACert(input *cert.CACSRInput) (*cert.Certificate, error) {
+	// prepare the certificate template
+	certTemplate := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
+		Subject: pkix.Name{
+			CommonName:         input.CommonName,
+			Country:            []string{input.Country},
+			Locality:           []string{input.Locality},
+			Organization:       []string{input.Organization},
+			OrganizationalUnit: []string{input.OrganizationUnit},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(input.Expiry),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	// generate key
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the certificate
+	caBytes, err := x509.CreateCertificate(rand.Reader, certTemplate, certTemplate, &caPrivKey.PublicKey, caPrivKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert Certificate into PEM format
+	caPEM := new(bytes.Buffer)
+	pem.Encode(caPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+
+	// convert Private Key into PEM format
+	caPrivKeyPEM := new(bytes.Buffer)
+	pem.Encode(caPrivKeyPEM, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(caPrivKey),
+	})
+
+	// create the clab certificate struct
+	clabCert := &cert.Certificate{
+		Cert: caPEM.Bytes(),
+		Key:  caPrivKeyPEM.Bytes(),
+	}
+
+	return clabCert, nil
+}
+
+// GenerateAndSignNodeCert generates and signs a node certificate, key and CSR based on the provided input and signs it with the CA.
+func (ca *CA) GenerateAndSignNodeCert(input *cert.NodeCSRInput) (*cert.Certificate, error) {
+	certTemplate := &x509.Certificate{
+		RawSubject:   []byte{},
+		SerialNumber: big.NewInt(1658),
+		Subject: pkix.Name{
+			Organization:       []string{input.Organization},
+			OrganizationalUnit: []string{input.OrganizationUnit},
+			CommonName:         input.CommonName,
+			Country:            []string{input.Country},
+			Locality:           []string{input.Locality},
+		},
+		DNSNames: input.Hosts,
+		// IPAddresses:  []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback},
```

Comment:
This is amazing stuff @steiler 

During my quick test I noticed that IP-based SAN's are not parsed correctly?

This commented line is likely the reason I see IP addresses being encoded as DNS SANs, and not IPs:

```
            X509v3 Subject Alternative Name: 
                DNS:srl, DNS:clab-srl-srl, DNS:srl.srl.io, DNS:172.20.20.4, DNS:2001:172:20:20::4
```

Looks like we need to have a function that iterates over `input.Hosts` and checks if an IP is passed and populate IPAddresses slice with them?

Reply:
Did that in fbd941c